### ISSUE TITLE
Compatibility of Filename on Windows

### ISSUE THREAD

**Issue by yakobowski** [janestreet/core_kernel#88]
I noticed that the `Filename` component of Core has been (mostly) migrated into Core_kernel. However, while trying to use it under Windows, I noticed that it is actually _not_ portable at all. In particular, forward slashes are expected everywhere. Fixing this requires actually pretty deep changes. Are you aware of this issue?

**Comment by yminsky** [comment]
Indeed. That module in its current state would perhaps be better named Posix_filename. We'd be happy to consider PRs on fixing this to make it more portable, if people would be interested in contributing. We don't have a lot of in-house expertise on using OCaml on Windows, so we're not well placed to get the semantics of filenames just right.

**Comment by yminsky** [comment]
And, it's worth saying, a comment explaining the posix-specific nature of the module would be in order. That's certainly something we can do.

**Comment by yakobowski** [comment]
I'm unfortunately not an expert either... I noticed that Daniel has done a lot of work related to filenames in Fpath (https://github.com/dbuenzli/fpath), this may be the way to go to get something portable.

In the meantime, a comment regarding portability would be great indeed :smile:

**Comment by dra27-js** [comment]
Fixed (finally!) and being pushed shortly.
